components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -421,49 +462,30 @@ void BraveReferralsService::PerformFinalizationChecks() {
   base::PostTask(
```

Comment:
not sure why these are separate PostTask here so I just left them as-is with WeakPtr